### DISCUSSION TITLE
Channelreader: Get id of sending channel in source.

### DISCUSSION THREAD

**Discussion by Kuepp1** [nextgenhealthcare/connect#5819] [General]
Hello lovely people,

i have a channel that awaits input from other channels in its source channelreader.

Like:

ChannelA
  Source: Filereader
  Destination: Channelwriter
ChannelB
  Source: Filereader
  Destination: Channelwriter

ChannelD, ChannelE, ChannelF and so on

They all sennd to ChannelX which then sends to a TCP Destination

ChannelX
  Source: Channelreader
  Destination: TCP Sender

Now i would like to have a dashboard display of which channel did send the message to ChannelX

How could i do that?

Thank's for helping me out!

**Comment by jonbartels** [comment]
https://docs.nextgen.com/bundle/Mirth_User_Guide_40/page/connect/connect/topics/c_Channel_Writer_connect_ug.html

Create a map variable called `channelName` and set it to the channel name. Then add `channelName` to the metadata table. It will pass that value as a source map variable to the receiving channel. 

I also suggest passing `messageId`. That way when you look at Channel X it is easy to see that Channel A Message ID #### was the origin of the message that hit channel X.

**Reply by Kuepp1** [reply]
That sounds easy. With "Create a map variable" you mean just add a variable to the channelMap right? And then add that to the metadata of the channelwriter. Would be sendingChannelName in my case.

And then i should retrieve it in my destination channel via sourceMap.get('sendingChannelName')?

**Reply by pacmano1** [reply]
Just a note, if you are using a channel writer then the source channel ID and message ID are already in the source map for the other channel.   The channel name is not though:

<img width="964" alt="image" src="https://github.com/nextgenhealthcare/connect/assets/44065187/0c8f7f84-b2ed-4c7d-a5f0-fca77a7b155e">

**Reply by Kuepp1** [reply]
Oh that's helpful. Thank you!

**Comment by Kuepp1** [comment]
Alright it works. Thank you guys for your help.

**Reply by jonbartels** [reply]
Mark an answer as accepted. That helps someone with a similar question find a solution
